source/agora/consensus/protocol/Nominator.d

```diff
@@ -577,7 +578,14 @@ extern(D):
             return;  // slot was already externalized or envelope is too new
         }
 
-        const PublicKey public_key = PublicKey(envelope.statement.nodeID[]);
+        const Hash utxo = Hash(envelope.statement.nodeID[][0 .. Hash.sizeof]);
+        UTXO utxo_value;
+        if (!this.ledger.peekUTXO(utxo, utxo_value))
+        {
+            log.trace("Missing UTXO: {} at height {}", utxo, last_block.header.height);
```

Comment:
I think the message should be more particular. It might just be that the validator is not enrolled!
```suggestion
            log.trace("Couldn't find UTXO {} at height {} to validate envelope's signature", utxo, last_block.header.height);
```